### ISSUE TITLE
Problem on downloading jlibgphoto2

### ISSUE THREAD

**Issue by xellosdark** [mvmn/gp2srv#7]
=== Error on downloading
Downloading: https://repo.maven.apache.org/maven2/x/mvmn/jlibgphoto2/jlibgphoto2/1.7/jlibgphoto2-1.7.pom
[WARNING] The POM for x.mvmn.jlibgphoto2:jlibgphoto2:jar:1.7 is missing, no dependency information available

I have tried to open the web site https://repo.maven.apache.org/maven2/x and this directory does not exist anymore.

Please advice the new site and thanks

**Comment by mvmn** [comment]
As mentioned in readme, dependencies are:
> Dependencies:
> - LibGPhoto2 - http://www.gphoto.org
> - Java 1.5 and above
> - https://github.com/mvmn/jlibgphoto2

So you need to clone jlibgphoto2 and do maven clean install in it.
Note that jlibgphoto2 in it's turn depends on https://github.com/mvmn/gphoto2-jna , so the process is this:
- Clone https://github.com/mvmn/gphoto2-jna
- Do maven clean install for gphoto2-jna
- Clone https://github.com/mvmn/jlibgphoto2
- Do maven clean install for jlibgphoto2
- Clone and build gp2srv

Or you could just take a pre-made release of gp2srv from [releases](https://github.com/mvmn/gp2srv/releases) - the ZIP files contain all dependencies JARs in lib folder.

**Comment by xellosdark** [comment]
Thanks and let me try again

Regards,
  Francis


在 2017年4月17日，上午4:14，Mykola Makhin <notifications@github.com<mailto:notifications@github.com>> 写道：


As mentioned in readme, dependencies are:

Dependencies:

  *   LibGPhoto2 - http://www.gphoto.org
  *   Java 1.5 and above
  *   https://github.com/mvmn/jlibgphoto2

So you need to clone jlibgphoto2 and do maven clean install in it.
Note that jlibgphoto2 in it's turn depends on https://github.com/mvmn/gphoto2-jna , so the process is this:

  *   Clone https://github.com/mvmn/gphoto2-jna
  *   Do maven clean install for gphoto2-jna
  *   Clone https://github.com/mvmn/jlibgphoto2
  *   Do maven clean install for jlibgphoto2
  *   Clone and build gp2srv

Or you could just take a pre-made release of gp2srv from releases<https://github.com/mvmn/gp2srv/releases> - the ZIP files contain all dependencies JARs in lib folder.

**Comment by xellosdark** [comment]
Sorry for keep asking.

Error is reported for "Unsupported major.minor version 52.0"
Checked on google.com, and the root is claimed to be unsupported java version; however, the version of java used is 1.7.0_121 (raspberry PI)

any clue?

Regards,
  Francis 
=====
pi@fl-raspi:~/Projects/gp2srv-2.7 $ java -version
java version "1.7.0_121"
OpenJDK Runtime Environment (IcedTea 2.6.8) (7u121-2.6.8-2~deb8u1+rpi1)
OpenJDK Zero VM (build 24.121-b00, interpreted mode)

=====
pi@fl-raspi:~/Projects/gp2srv-2.7 $ ./run.sh
Exception in thread "main" java.lang.UnsupportedClassVersionError: org/eclipse/jetty/server/handler/ErrorHandler : Unsupported major.minor version 52.0
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:803)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:442)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:64)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:354)
        at java.net.URLClassLoader$1.run(URLClassLoader.java:348)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.net.URLClassLoader.findClass(URLClassLoader.java:347)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        at x.mvmn.gp2srv.GPhoto2ServerLauncher.main(GPhoto2ServerLauncher.java:63)

**Comment by mvmn** [comment]
https://en.wikipedia.org/wiki/Java_class_file

> Java SE 9 = 53 (0x35 hex),
Java SE 8 = 52 (0x34 hex),
Java SE 7 = 51 (0x33 hex),
Java SE 6.0 = 50 (0x32 hex),
Java SE 5.0 = 49 (0x31 hex),
JDK 1.4 = 48 (0x30 hex),
JDK 1.3 = 47 (0x2F hex),
JDK 1.2 = 46 (0x2E hex),
JDK 1.1 = 45 (0x2D hex).

You've got something compiled for Java 8 (class file has version 52 - which stands for Java 8, see table above), but try to use it with Java 7 (as you've mentioned yourself, you've got Java 1.7.0_121)

**Comment by mvmn** [comment]
Please use Java8 for GPhoto2Server.

If you must use older version of Java, please use compatibility branch:
https://github.com/mvmn/gp2srv/releases/tag/J5_2.7
